File: camera_capture.cpp
#include <sl/Camera.hpp>
#include <iostream>
#include <cstdint>
#include <cstdlib>
#include <filesystem>
#include <string>

using namespace std;
using namespace sl;

int main(int argc, char** argv) {
    Camera zed;

    InitParameters init_params;
    init_params.camera_resolution = RESOLUTION::AUTO;
    init_params.camera_fps = 15;
    init_params.depth_mode = DEPTH_MODE::NEURAL_LIGHT;

    ERROR_CODE err = zed.open(init_params);
    if (err > ERROR_CODE::SUCCESS) {
        cout << "Error " << err << ", exit program." << endl;
        return EXIT_FAILURE;
    }

    std::filesystem::create_directories("camera_test");

    int i = 0;
    sl::Mat image;
    RuntimeParameters runtime_parameters;

    int64_t lastTime = 0;

    while (i < 1000) {
        if (zed.grab(runtime_parameters) <= ERROR_CODE::SUCCESS) {
            zed.retrieveImage(image, VIEW::LEFT);

            Timestamp timestamp = zed.getTimestamp(TIME_REFERENCE::IMAGE);
            int64_t curTime = timestamp.getMilliseconds();

            cout << (curTime - lastTime) << endl;
            lastTime = curTime;

            string filename = "camera_test/" + to_string(curTime) + ".png";

            ERROR_CODE write_err = image.write(filename.c_str());
            if (write_err > ERROR_CODE::SUCCESS) {
                cout << "Failed to save: " << filename << endl;
            }

            i++;
        }
    }

    zed.close();
    return EXIT_SUCCESS;
}

File: camera_capture.py
import pyzed.sl as sl
import os
import cv2
import numpy as np


def take_photo():

   zed = sl.Camera()
   init_params = sl.InitParameters()
   init_params.camera_resolution = sl.RESOLUTION.AUTO 
   init_params.camera_fps = 15
   init_params.depth_mode = sl.DEPTH_MODE.NEURAL_LIGHT
   err = zed.open(init_params)
   if (err > sl.ERROR_CODE.SUCCESS) :
      exit(-1)

   i = 0
   image = sl.Mat()
   runtime_parameters = sl.RuntimeParameters()
   os.makedirs("camera_test", exist_ok=True)
   lastime = 0
   while i < 1000:
      if zed.grab(runtime_parameters) <= sl.ERROR_CODE.SUCCESS:
         zed.retrieve_image(image, sl.VIEW.LEFT)

         timestamp = zed.get_timestamp(sl.TIME_REFERENCE.IMAGE)
         cur_time = timestamp.get_milliseconds()

         print(cur_time - lastime)
         lastime = cur_time

         filename = f"camera_test/{cur_time}.png"
         err = image.write(filename)
         if err != sl.ERROR_CODE.SUCCESS:
               print("failed to save", filename, err)

         i += 1
           

   zed.close()


take_photo()
